### PR TITLE
[RNMobile] Performance improvements in File related controls

### PR DESCRIPTION
**Pull request by lukewalczak**
**Merge after `Video` PR: https://github.com/WordPress/gutenberg/pull/28220**

## Description

Part of: https://github.com/WordPress/gutenberg/pull/28196

Contains changes only related to the `File` block and controls available within block settings and inherit changes from `Video` PR.

| BLOCK			| KIND OF SETTING		| CONTROL COMPONENT		| NESTED CONTROL COMPONENT	|
| :---			| :---: 				|  :---: 				| :---:  					|
| File			| File block Settings	| `SelectControl`		| 							|
|				|						| `ToggleControl`		| 							|	
|				|						| `ToggleControl`		| 							|	
|				|						| `TextControl`			| 							|	

## How has this been tested?

Test manually `File` block on web and mobile.

You can paste the snippet to have `File` block filled:

```
<!-- wp:file {"id":2620,"href":"https://callstackhappiryuhome.files.wordpress.com/2020/12/pexels-photo-3732527.jpg"} -->
<div class="wp-block-file"><a href="https://callstackhappiryuhome.files.wordpress.com/2020/12/pexels-photo-3732527.jpg">pexels-photo-3732527</a><a href="https://callstackhappiryuhome.files.wordpress.com/2020/12/pexels-photo-3732527.jpg" class="wp-block-file__button" download>Download</a></div>
<!-- /wp:file -->
```

_or_

1. Open post on web (run gutenberg locally)
2. Add `File` block and attach file
3. Play with settings and observe if work properly
4. Enable all of them and save post
5. Open post on mobile
6. Play with settings

**Expect everything works correctly**

### RESULTS

**1. pressing copy url button**

- updating time reduced from **7.06ms** to **3.49ms**
- avoid redundant updates in other controls

before | after
--- | ---
Screenshot 2021-01-15 at 11 20 33| Screenshot 2021-01-15 at 11 19 33

## Types of changes

Performance improvements

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation. 
- [X] I've included developer documentation if appropriate. 
- [X] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/text-control/index.native.js [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
+				value={ value }
+				onChangeValue={ onChange }
+				aria-describedby={ !! help ? id + '__help' : undefined }
+				{ ...props }
+			/>
+		);
+	}
+);
 
 export default TextControl;
```
Comment (dratwas) [inline packages/components/src/text-control/index.native.js:41]: WDYT about using memo here? `export default memo(TextControl)`?

**Review by dratwas — APPROVED** [review]

**Comment by dcalhoun** [comment]
@lukewalczak while working on unrelated work, I noticed that this PR [removes the `disabledButton` class name](https://github.com/WordPress/gutenberg/commit/ccc1e86cf740e3bce2f54c1685db68709713e474#diff-5efce7c6e9fac1465bdeeecb03388d1f69272d182279ceb5a9a5ef28c15113f7L21), but there is a [lingering reference to it](https://github.com/WordPress/gutenberg/blob/b13f304c2a42d9c3bcf1c6650062fb73c22fcb11/packages/block-library/src/file/edit.native.js#L435) for the download button. From debugging, this lingering reference should control the styling of the download button that is populated after adding a file.

I presume this likely an oversight of some kind. Do you know if the intention was to keep the lowered opacity for this button when it's disabled or was there a different styling that should be applied? Either way, we likely need to fix/remove the lingering reference to a non-existing class name.

<details>
<summary>Reference Image</summary>

disabled-download-button-state

</details>

Once I understand the desired outcome, I'm happy to help create a new issue to track the work. Thanks!

**Comment by jd-alexander** [comment]
> @lukewalczak while working on unrelated work, I noticed that this PR [removes the `disabledButton` class name](https://github.com/WordPress/gutenberg/commit/ccc1e86cf740e3bce2f54c1685db68709713e474#diff-5efce7c6e9fac1465bdeeecb03388d1f69272d182279ceb5a9a5ef28c15113f7L21), but there is a [lingering reference to it](https://github.com/WordPress/gutenberg/blob/b13f304c2a42d9c3bcf1c6650062fb73c22fcb11/packages/block-library/src/file/edit.native.js#L435) for the download button. From debugging, this lingering reference should control the styling of the download button that is populated after adding a file.
> 
> 
> 
> I presume this likely an oversight of some kind. Do you know if the intention was to keep the lowered opacity for this button when it's disabled or was there a different styling that should be applied? Either way, we likely need to fix/remove the lingering reference to a non-existing class name.
> 
> 
> 
> <details>
> 
> <summary>Reference Image</summary>
> 
> 
> 
> disabled-download-button-state
> 
> 
> 
> </details>
> 
> 
> 
> Once I understand the desired outcome, I'm happy to help create a new issue to track the work. Thanks!

Hi @dcalhoun I just happened to see this comment. I worked on the File Block so I can confirm that the opacity should be lowered when the Download button is disabled. One of the states in which it gets disabled is during the file upload process.

**Comment by lukewalczak** [comment]
Looks like the oversight from my side. If you can please create an issue for that and I will handle it asap to have the fix upcoming release ✌🏽 Thank you!

**Comment by lukewalczak** [comment]
Created a [PR](https://github.com/WordPress/gutenberg/pull/28704) which moves back the style. Could you please @jd-alexander or @dcalhoun do a testing and reviewing?

**Comment by dcalhoun** [comment]
@jd-alexander @lukewalczak thanks for the info. https://github.com/WordPress/gutenberg/issues/28706 captures the issue. I will check out the PR branch shortly.

**Pull request merged**